Zend/zend_language_scanner.l

```diff
@@ -1600,9 +1616,11 @@ NEWLINE ("\r"|"\n"|"\r\n")
 }
 
 <ST_IN_SCRIPTING>{BNUM} {
-	char *bin = yytext + 2; /* Skip "0b" */
+	/* The +/- 2 skips "0b" */
 	int len = yyleng - 2;
-	char *end;
+	char *end, *bin = strndup(yytext + 2, len);
```

Comment:
Where is this freed again? Also, should use `estrndup`. It would also be good point to avoid copying the string if it does not have digit separators.